### PR TITLE
LESQ-1109: year group search filters

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 1986

- Wire in changes to search api into OWA
- Plug in mv_search_page_10_0_0 into OWA - which should remove redundant search filters https://www.notion.so/oaknationalacademy/I-want-to-remove-redundant-subject-filters-on-search-11b26cc4e1b180d4a879c53ff2a06302
- Add year group search filters to search page
- Update active search filters when year group filter selected
- Update corresponding tests and schema definitions

## Issue(s)

Fixes #LESQ-1109

### AC

- [ ] Update search api to enable filtering by year
- [ ] In OWA update search to enable filtering by year
- [ ] If it is straightforward to exclude ‘Reception’ then do so, otherwise leave it (as we still get EYFS as a keystage anyway)
- [ ] Update the search result cards to display year

## How to test

1. Go to https://deploy-preview-2929--oak-web-application.netlify.thenational.academy/teachers/search
2. Click on any year group filter
3. You should see lessons and units have the associated year group with that filter

## Screenshots

How it should look:

https://github.com/user-attachments/assets/d0a3fe64-de39-46c8-9dba-ca8758fba689

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/index.ts [hunk 1]
```diff
@@ -48,22 +48,7 @@ const teachersHomePageData = z.object({
...
-  slug: z.string(),
-  displayOrder: z.number().optional(),
-});
-export const ks4OptionSchema = z.object({
-  title: z.string(),
-  slug: z.string(),
-  displayOrder: z.number().optional(),
-});
-export const keystagesSchema = z.object({
+const genericFilteringGroup = z.object({
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/index.ts:51]: I struggled with a name for this

File: src/node-lib/curriculum-api-2023/queries/searchPage/searchPage.query.ts [hunk 2]
```diff
@@ -14,10 +14,6 @@ const searchPageQuery = (sdk: Sdk) => async () => {
 
   return {
     ...parsed,
-    subjects: parsed.subjects.filter(
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/searchPage/searchPage.query.ts:17]: This is now being removed at the mv level

File: src/node-lib/curriculum-api-2023/index.ts [hunk 3]
```diff
@@ -48,22 +48,7 @@ const teachersHomePageData = z.object({
   keyStages: z.array(keyStageSchema),
 });
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/index.ts:50]: Naming is tough 

Some of this potentially could be extrapolated to the curriculum schema lib?

**Comment by RBrownsett** [comment]
Tested, works as expected, years group filter shows the correct results. However I just want to check a couple of issues:

@k-huggs just checking is it possible to search for a EYFS lesson? When I enter a lesson from reception e.g. 'A problem shared is a problem halved' there are no results for that lesson (https://www.thenational.academy/teachers/programmes/personal-social-and-emotional-development-foundation-early-years-foundation-stage-l/units/new-beginnings-5016/lessons/a-problem-shared-is-a-problem-halved-6nhp6d)

Also I noticed that the subject list filter was different to live (see screenshot below - with missing subjects such as English grammar, Expressive art and design) is that intentional?

LIVE:
Screenshot 2024-10-25 at 12 33 03

PR:
image

**Comment by k-huggs** [comment]
> Tested, works as expected, years group filter shows the correct results. However I just want to check a couple of issues:
> 
> @k-huggs just checking is it possible to search for a EYFS lesson? When I enter a lesson from reception e.g. 'A problem shared is a problem halved' there are no results for that lesson ([thenational.academy/teachers/programmes/personal-social-and-emotional-development-foundation-early-years-foundation-stage-l/units/new-beginnings-5016/lessons/a-problem-shared-is-a-problem-halved-6nhp6d](https://www.thenational.academy/teachers/programmes/personal-social-and-emotional-development-foundation-early-years-foundation-stage-l/units/new-beginnings-5016/lessons/a-problem-shared-is-a-problem-halved-6nhp6d))
> 
> Also I noticed that the subject list filter was different to live (see screenshot below - with missing subjects such as English grammar, Expressive art and design) is that intentional?
> 
> LIVE: Screenshot 2024-10-25 at 12 33 03
> 
> PR: image

Yeah those changes are both intentional, see this ticket here https://www.notion.so/oaknationalacademy/I-want-to-remove-redundant-subject-filters-on-search-11b26cc4e1b180d4a879c53ff2a06302 This removes the specialist, eyfs and additional english subjects from the filters

**Review by RBrownsett — APPROVED** [review]
Approved - tested in desktop and mobile. Tried different combinations of filter, results returned as as expected. Voice over works the same as previous filters, informing the user of the contents of the search results.

**Pull request merged**